### DISCUSSION TITLE
How to fetch data per minute with server components/actions?

### DISCUSSION THREAD

**Discussion by Stupidism** [vercel/next.js#60631] [App Router] [answered]
My current implementation is to use client components:

```

  const {
    data,
    isFetching,
    dataUpdatedAt,
  } = useSuspenseQuery({
    queryKey: ['jobs', jobId, 'candidates'],
    queryFn: () => getMyData(), // this is a client-side api call
    refetchInterval: (query) => isDataOutdated(query.state) && 1000,
  });
```

Is there a decent way to do this with server actions?

**Comment by nirvikpurkait** [comment, marked as answer]
You can fetch data in server component using `fetch`, if you want to fetch data per minute, that means you want to validate the data if it is correct or not, you can achieve that using `fetch` along with `{ next: { revalidate: time_in_second } }` .

For more info you can check [here](https://nextjs.org/docs/app/building-your-application/data-fetching/fetching-caching-and-revalidating#time-based-revalidation)

**Reply by Stupidism** [reply]
Thanks! You saved my life~~

**Reply by Stupidism** [reply]
@nirvikpurkait How about scroll to load more? How to do such a thing with server actions?
